web/src/components/vis/VolcanoPlot.vue

```diff
@@ -1,16 +1,39 @@
-<script>
+<script lang="ts">
+import {
+  PropType, defineComponent, computed, ref, watchEffect, onMounted, Ref,
+} from '@vue/composition-api';
+import { axisBottom, axisLeft } from 'd3-axis';
+import { scaleLinear } from 'd3-scale';
 import { select } from 'd3-selection';
+import useAxisPlot from './use/useAxisPlot';
 
-import { axisPlot } from './mixins/axisPlot';
+interface Row {
+  pValue: number;
+  log2FoldChange: number;
+  name: string;
+  color?: string;
+}
 
-export default {
-  mixins: [
-    axisPlot,
-  ],
+interface TransformedRow extends Row {
+  x: number;
+  y: number;
+}
```

Comment:
I like the use of `extends` to describe what additional properties decorate the regular data.